src/orion/service/broker/broker.py

```diff
@@ -0,0 +1,235 @@
+"""Brokers handle the request to the orion code,
+they are the bridge between the API and orion python
+
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+import falcon
+
+from orion.client import ExperimentClient, build_experiment
+from orion.core.io.database.mongodb import MongoDB
+from orion.storage.legacy import Legacy
+
+log = logging.getLogger(__name__)
+
+
+def success(values) -> dict:
+    """Create a success response"""
+    return dict(status=0, result=values)
+
+
+def error(exception) -> dict:
+    """Create an error response"""
+    return dict(status=1, error=str(exception))
+
+
+@dataclass
+class Database:
+    """Database configuration"""
+
+    database: str = "orion"
+    host: str = "localhost"
+    port: int = 8123
+
+
+@dataclass
+class Authentication:
+    """Authentication configuration"""
+
+    database: str = "users"
+    host: str = "localhost"
+    port: int = 8124
+    usernane: str = "god"
+    password: str = "god123"
+
+
+@dataclass
+class ServiceContext:
+    """Global configuration for the service"""
+
+    database: Database = Database()
+    authentication: Authentication = Authentication()
+    broker: ExperimentBroker = None
+    auth = AuthenticationService = None
+
+
+@dataclass
+class RequestContext:
+    """Request specific information useful throughout the request handling"""
+
+    service: ServiceContext  # Service config
+    username: str  # Populated on authentication
+    password: str
+    data: dict  # Parameter data from the request
+    token: str  # Original token
+    request: falcon.Request = None  # Original Request
+    response: falcon.Response = None  # Response to be sent back
+
+
+def get_storage_for_user(request: RequestContext):
+    """Create a mongodb connection for a given user"""
+    log.debug("Connecting to database")
+
+    assert request.username is not None
+
+    log.debug(
+        "Mongo %s, %s, %s",
+        request.service.database.host,
+        request.service.database.port,
+        request.service.database.database,
+    )
+
+    db = MongoDB(
+        name=request.service.database.database,
+        host=request.service.database.host,
+        port=request.service.database.port,
+        username=request.username,
+        password=request.password,
+        owner=request.username,
+    )
+
+    storage = Legacy(
+        database_instance=db,
+        # Skip setup, this is a shared database
+        # we might not have the permissions to do the setup
+        # and the setup should already be done anyway
```

Comment:
When would the setup be done? We would need a script or command to do this when we set in place a shared database.

Reply:
Setup is done by IT when the database is created, `setup.sh` has a template of the code that could be executed to setup the database